Mathlib/LinearAlgebra/TensorProduct/LinearMap.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2024 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import Mathlib.LinearAlgebra.Dual
+import Mathlib.LinearAlgebra.TensorProduct.Tower
+
+/-!
+# Bilinear maps on a Tensor Product
+
+## Main definitions
+
+* `LinearMap.tensorDistrib (B₁ ⊗ₜ B₂)`: the bilinear map on `M₁ ⊗ M₂` constructed by applying
+  `B₁` on `M₁` and `B₂` on `M₂`.
+
+## Implementation notes
+
+Bilinear map versions of some results in `LinearAlgebra.BilinearForm.TensorProduct`.
```

Comment:
Can you prove one using the other, rather than copy-pasting the file?

Reply:
I still think we should get rid of `BilinForm` and tangling the replacement to the deprecated definition only makes everything worse.

Reply:
In this case, I'm suggesting we reprove the `BilinForm` case in terms of the `LinearMap` case.

Reply:
(I agree in the long term `structure BilinForm` should not exist)

Reply:
@eric-wieser To define `BilinForm.tensorDistrib` in terms of `LinearMap.tensorDistrib` I think we need a map:
```
BilinForm A M₁ ⊗[R] BilinForm R M₂ →ₗ[A] (M₁ →ₗ[A] M₁ →ₗ[A] A) ⊗[R] (M₂ →ₗ[R] M₂ →ₗ[R] R)
```
This is okay:
```
#check TensorProduct.map ((toLin (R₂ := R) (M₂ := M₁)).toLinearMap )  ((toLin (R₂ := R) (M₂ := M₂)).toLinearMap)
```
But `TensorProduct.lift` is not hetrobasic, so this fails:
```
#check TensorProduct.map ((toLin (R₂ := A) (M₂ := M₁)).toLinearMap )  ((toLin (R₂ := R) (M₂ := M₂)).toLinearMap)
```

Reply:
I'm pretty sure the current definition of `BilinForm.tensorDistrib` already contains the new definition as a subexpression?

Reply:
Ah!
```
#check TensorProduct.AlgebraTensorModule.congr ((toLin (R₂ := A) (M₂ := M₁))) (toLin (R₂ := R) (M₂ := M₂)) 
```
Thanks.

Reply:
So, something like this?
```
def tensorDistrib : BilinForm A M₁ ⊗[R] BilinForm R M₂ →ₗ[A] BilinForm A (M₁ ⊗[R] M₂) :=
  LinearMap.toBilin.toLinearMap ∘ₗ (LinearMap.tensorDistrib R A ∘ₗ 
    (TensorProduct.AlgebraTensorModule.congr toLin toLin ).toLinearMap)
```

Reply:
Yes, that looks good to me